ipfsconn/ipfshttp/ipfshttp_test.go

```diff
@@ -99,24 +100,57 @@ func testPin(t *testing.T, method string) {
 	case "refs":
 		ipfs.config.PinTimeout = 1 * time.Second
 		c3 := test.SlowCid1
-		err = ipfs.Pin(ctx, c3, -1)
+		err = ipfs.Pin(ctx, api.PinCid(c3))
 		if err == nil {
 			t.Error("expected error pinning cid")
 		}
 	case "pin":
 		ipfs.config.PinTimeout = 5 * time.Second
 		c4 := test.SlowCid1
-		err = ipfs.Pin(ctx, c4, -1)
+		err = ipfs.Pin(ctx, api.PinCid(c4))
 		if err == nil {
 			t.Error("expected error pinning cid")
 		}
 	default:
 	}
 }
 
+func testPinUpdate(t *testing.T) {
+	ctx := context.Background()
+	ipfs, mock := testIPFSConnector(t)
+	defer mock.Close()
+	defer ipfs.Shutdown(ctx)
+
+	pin := api.PinCid(test.Cid1)
+	pin.PinUpdate = test.Cid1
+	// enforce pin/update even though it would be skipped
+	ipfs.config.PinMethod = "update"
+	err := ipfs.Pin(ctx, pin)
+	if err == nil {
+		t.Fatal("expected an error")
+	}
+	if !strings.HasSuffix(err.Error(), "recursively pinned already") {
+		t.Fatal("expected error about from not being pinned")
+	}
+
+	ipfs.config.PinMethod = "pin"
+	err = ipfs.Pin(ctx, pin)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// This should trigger the pin/update path
+	pin.Cid = test.Cid2
+	err = ipfs.Pin(ctx, pin)
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
 func TestIPFSPin(t *testing.T) {
 	t.Run("method=pin", func(t *testing.T) { testPin(t, "pin") })
 	t.Run("method=refs", func(t *testing.T) { testPin(t, "refs") })
+	t.Run("method=update", func(t *testing.T) { testPinUpdate(t) })
```

Comment:
here as well

Reply:
no, we are testing the case when we trigger a pin/update and not a pin.